src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java

```diff
@@ -60,7 +64,9 @@
   PERMANENT_LOAN_TYPE_ID("permanentLoanTypeId", "$.item.permanentLoanTypeId", LOAN_TYPES),
   STATUS("status", "$.item.status.name"),
   VOLUME("volume", "$.item.volume"),
-  YEARCAPTION("yearCaption", "$.item.yearCaption");
+  YEARCAPTION("yearCaption", "$.item.yearCaption"),
+  ITEM_NOTE_TYPE("itemNoteTypeId", "$.item.notes[?(@.itemNoteTypeId=={id}) && ?(!(@.staffOnly) || @.staffOnly == false))]", ITEM_NOTE_TYPES),
+  ITEM_NOTE_TYPE_STAFF_ONLY("itemNoteTypeId.staffOnly", "$.item.notes[?(@.itemNoteTypeId=={id}) && ?(!(@.staffOnly) || @.staffOnly == true))]", ITEM_NOTE_TYPES);
```

Comment:
this  rule is checking staff only **not present** for both ITEM_NOTE_TYPE rules and this one, this must be staffonly present and is true